pyclam/search.py

```diff
@@ -122,7 +130,7 @@ def _parse_query(self, query: Datum) -> np.ndarray:
         self._check_shape(query)
         return query
 
-    def tree_search(self, query: Datum, radius: float, start: Cluster, *, max_depth: Optional[int] = None) -> ClusterResults:
+    def tree_search(self, query: Datum, radius: float, *, start: Cluster = None, max_depth: int = None) -> ClusterResults:
```

Comment:
Hmm, I see you've trimmed a lot of `Optional`'s off of places that would happily except `None`, which seems very optional IMO. I would be very curious why you've made these changes!

Reply:
Giving a type-hint and then setting the default value to None means that at interpretation time, that time automatically gets an Optional put around it.